### PR TITLE
[stable9.1] Faster classification migration

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes https://github.com/owncloud/core/issues/25617
- [ ] apply same mechanism for birthday sync @deepdiver1975

@sbraz mind testing - thx
## Test Scenario
1. Run occ upgrade
2. see the progress on processed calendar objects

```
deepdiver@alien:~/Development/ownCloud/master [faster-classification-migration]$ ./occ up
ownCloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Set log level to debug
Turned on maintenance mode
Checking whether the database schema can be updated (this can take a long time depending on the database size)
Checked database schema update
Checking updates of apps
Checking whether the database schema for <dav> can be updated (this can take a long time depending on the database size)
 Done                                                        
 7/7 [============================] 100%                     
Checked database schema update for apps
Updating database schema
Updated database
Updating <dav> ...
Fix classification for calendar objects
 Done                                                        
 6/6 [============================] 100%                     
Updated <dav> to 0.2.7
Drop old database tables
 Done                                                        
 28/28 [============================] 100%                   
Remove old (< 9.0) calendar/contact shares
 Done                                                        
 4/4 [============================] 100%                     
Fix permissions so avatars can be stored again
 Done                                                        
 2/2 [============================] 100%                     
Update successful
Turned off maintenance mode
Reset log level
```

### CODE REVIEW

File: apps/dav/lib/Migration/Classification.php [hunk 1]
```diff
@@ -33,33 +32,18 @@ class Classification implements IRepairStep {
...
+	/** @var IDBConnection */
+	private $connection;
 
 	/**
 	 * Classification constructor.
 	 *
 	 * @param CalDavBackend $calDavBackend
 	 */
-	public function __construct(CalDavBackend $calDavBackend, IUserManager $userManager) {
+	public function __construct(CalDavBackend $calDavBackend, IDBConnection $connection) {
```
Comment (VicDeo) [inline apps/dav/lib/Migration/Classification.php]: @DeepDiver1975 PHPdoc is missing for the second argument

Comment (DeepDiver1975) [inline apps/dav/lib/Migration/Classification.php]: Thx. Feel free to add. Thx

**Comment by VicDeo** [comment]
:+1: Looks good

**Comment by PVince81** [comment]
@DeepDiver1975 @felixboehm backport to 9.0 ?

**Comment by PVince81** [comment]
- [ ] also need master/9.2 PR

**Comment by PVince81** [comment]
@DeepDiver1975 ping for the ports

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 ping for the ports

not needed - migration is only from 9.0 to 9.1

**Comment by butonic** [comment]
@DeepDiver1975 @PVince81 actually the same happens when migrating from OC8.2x to OC9.0x

**Comment by PVince81** [comment]
@DeepDiver1975 please backport

**Comment by butonic** [comment]
@DeepDiver1975 this should help with upgrades, right? can you backport?

**Comment by DeepDiver1975** [comment]
This is only relevant for upgrades to 9.1

**Pull request merged**